File: provable-sdk-js/src/prove.ts
/**
 * Prove data utilities
 */

import { keccak256, keccak256_str } from './hash';
import { prove_single_hash } from './api';
import type { ProveSingleHashResponse } from './types';
import type { ProveOptions } from './options';

/**
 * Prove data by computing its hash and calling Kayros API
 * @param data - Input data as Uint8Array
 * @param dataType - Optional data type identifier (defaults to "provable_sdk")
 * @returns Promise with the Kayros response
 */
export async function prove_data(
  data: Uint8Array,
  dataTypeOrOptions?: string | ProveOptions
): Promise<ProveSingleHashResponse> {
  const dataHash = keccak256(data);
  return await prove_single_hash(dataHash, dataTypeOrOptions);
}

/**
 * Prove string data by computing its hash and calling Kayros API
 * @param str - Input string
 * @param dataType - Optional data type identifier (defaults to "provable_sdk")
 * @returns Promise with the Kayros response
 */
export async function prove_data_str(
  str: string,
  dataTypeOrOptions?: string | ProveOptions
): Promise<ProveSingleHashResponse> {
  const dataHash = keccak256_str(str);
  return await prove_single_hash(dataHash, dataTypeOrOptions);
}

File: provable-sdk-py/provable_sdk/prove.py
"""
Prove data utilities
"""

from .hash import keccak256, keccak256_str
from .api import prove_single_hash
from .types import ProveSingleHashResponse


def prove_data(data: bytes) -> ProveSingleHashResponse:
    """
    Prove data by computing its hash and calling Kayros API

    Args:
        data: Input data as bytes

    Returns:
        The Kayros response
    """
    data_hash = keccak256(data)
    return prove_single_hash(data_hash)


def prove_data_str(s: str) -> ProveSingleHashResponse:
    """
    Prove string data by computing its hash and calling Kayros API

    Args:
        s: Input string

    Returns:
        The Kayros response
    """
    data_hash = keccak256_str(s)
    return prove_single_hash(data_hash)
